src/components/business/SetPassword.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+
+import { SetPasswordError } from "@lib/@types/Password";
+import { User } from "@lib/@types/ProviderUser";
+import { HttpVerbs } from "@src/@types/HttpVerbs";
+import { ErrorSettingPassword } from "@src/clientErrors";
+import { fetchJson } from "@src/utils/fetchJson";
+
+interface SetPasswordProps {
+  children: (props: {
+    setPassword: (passwordInput: string) => Promise<SetPasswordOutput>;
+  }) => JSX.Element;
+}
+
+export type SetPasswordOutput = {
+  data: { createOrUpdatePassword: User };
+  restrictionRulesError?: SetPasswordError;
+};
+
+export const SetPassword: React.FC<SetPasswordProps> = ({ children }) => {
+  async function setPassword(
+    passwordInput: string,
+  ): Promise<SetPasswordOutput> {
+    return fetchJson("/api/auth-connect/set-password", HttpVerbs.POST, {
+      passwordInput,
+    }).then((response) => {
+      if (response.status >= 400 && response.status !== 422) {
```

Comment:
What is supposed to happen when it's a 422?

Reply:
When the `statusCode` is equal to `422`, the response is let through the next process, and catch is here:

```ts
// SetPasswordForm.tsx
if (restrictionRulesError) {
	setPasswordRestrictionError(restrictionRulesError.rules);

	setIsNotSubmitted(true);
}
```

It renders the Provider's password rules, and allows the user to submit again.